packages/core/src/state-management/repository.ts

```diff
@@ -166,28 +166,24 @@ export class Repository {
   }
 
   /**
-   * Sets the StateStore to use.
+   * Sets the KVFactory to use.
    * This must be called before init().
-   * @param stateStore
    */
-  async injectKeyValueStore(stateStore: IKVStore): Promise<void> {
+  injectKVFactory(factory: IKVFactory) {
     this.setDeps({
       ...this.#deps,
-      keyValueStore: stateStore,
+      kvFactory: factory,
     })
   }
 
   async init(): Promise<void> {
-    await this.#deps.keyValueStore.init()
-    await this.pinStore.open(this.#deps.keyValueStore)
-    await this.anchorRequestStore.open(this.#deps.keyValueStore) // Initialization hell
+    await this.pinStore.open(this.#deps.kvFactory)
+    await this.anchorRequestStore.open(this.#deps.kvFactory) // Initialization hell
```

Comment:
I know this comment was here before this PR but why is this `Initialization hell`?

Reply:
It was and is unclear in what order what store should be opened or closed hence "hell".